pkg/logger/zap.go

```diff
@@ -2,45 +2,62 @@ package logger
 
 import (
 	"fmt"
+	"os"
 	"time"
 
 	"github.com/go-logr/zapr"
+	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+// InitZapArgs represents a set of arguments for initializing the zap logger.
+type InitZapArgs struct {
+	Level          int
+	OutputFilePath string
+	Opts           []LoggerOpt
+}
+
 // InitZap creates a zap logger with the provided verbosity level
 // and sets it as the package logger.
 // 0 is the least verbose and 10 the most verbose.
 // The package logger can only be init once, so subsequent calls to this method
 // won't have any effect.
-func InitZap(level int, opts ...LoggerOpt) error {
-	cfg := zap.NewDevelopmentConfig()
-	cfg.Level = zap.NewAtomicLevelAt(zapcore.Level(-1 * level))
+func InitZap(args InitZapArgs) error {
```

Comment:
I feel like all of this would be easier to test if we split this method in
`InitZap` and `newZap`
`InitZap` would just call `newZap` and `set`.
And `newZap` would have the construction logic. That way you can test `newZap` directly and forget about `config`, `build`, etc. which are just implementation details.